### PR TITLE
Update documentation calib14

### PR DESCRIPTION
**Pull request by MSECode**
Update calibration14 documentation with procedure to follow for finding correct values of calibration parameters to be set in the configuration files

### CODE REVIEW

File: docs/icub_r1_icub3_calibration_types/icub_r1_calibration_types.md [hunk 1]
```diff
@@ -168,6 +168,15 @@ This calibration is used for joint with absolute encoder, it is the newest versi
 ### Calibration 14
 This calibration is used for joints with the absolute encoder at the joint (which currently works with POS service / FAP encoder sensor) and with incremental encoder at the motor. Moreover, it is applied to joints whose motor is provided with hard-stop mechanical limits. Specifically, this type of calibration is used when the user needs to reach one of the two mechanical limits on the joint motor, aiming at calibrating the incremental encoder at the motor and accurately setting the hardware limits between which the rotor can move.
 
+Aiming at helping the common user at finding the correct values for the paramters reported in the table below, which should be set in the calibration configuration file, the steps to follow for the identification of those values are here reported.
+First of all, the user should set the boolean values `skipCalibration` and `useRawEncoderData` in the file `general.xml` to `true`, therefore the calibration procedure won't start at `yarprobotinterface` startup and secondly it would be possible to allow the port `/ergocub/left_arm/state:o` to print the raw (i.e. uncalibrated values) for the finger joint encoders.
```
Comment (pattacini) [inline docs/icub_r1_icub3_calibration_types/icub_r1_calibration_types.md:172]:
```suggestion
First, set to `true` the boolean variables `skipCalibration` and `useRawEncoderData` in the file `general.xml`. This is meant to disable calibration at `yarprobotinterface` startup. Also, this will make the port `/ergocub/left_arm/state:o` print the raw values (i.e., uncalibrated) for the finger joint encoders. 
```

File: docs/icub_r1_icub3_calibration_types/icub_r1_calibration_types.md [hunk 2]
```diff
@@ -168,6 +168,15 @@ This calibration is used for joint with absolute encoder, it is the newest versi
...
 
+Aiming at helping the common user at finding the correct values for the paramters reported in the table below, which should be set in the calibration configuration file, the steps to follow for the identification of those values are here reported.
+First of all, the user should set the boolean values `skipCalibration` and `useRawEncoderData` in the file `general.xml` to `true`, therefore the calibration procedure won't start at `yarprobotinterface` startup and secondly it would be possible to allow the port `/ergocub/left_arm/state:o` to print the raw (i.e. uncalibrated values) for the finger joint encoders. 
+Done that, the user should check the values printed out by the port running in a terminal the command:
```
Comment (pattacini) [inline docs/icub_r1_icub3_calibration_types/icub_r1_calibration_types.md:173]:
```suggestion
Once done, you should check the values printed out by the port running in a terminal the command:
```

File: docs/icub_r1_icub3_calibration_types/icub_r1_calibration_types.md [hunk 3]
```diff
@@ -168,6 +168,15 @@ This calibration is used for joint with absolute encoder, it is the newest versi
...
+
+`yarp read ... /ergocub/left_arm/state:o` 
+
+and by doing this observing which should be the raw value of the offset for the encoder joint when the finger is at the hard-stop position (ideally finger full open). Thus, the user should set this value (with opposite sign) as the value for the parameter `calibration5`.
```
Comment (pattacini) [inline docs/icub_r1_icub3_calibration_types/icub_r1_calibration_types.md:177]:
```suggestion
While doing so, pay attention to the raw value of the offset of the encoder joint when the finger is at the hard-stop position (ideally, the finger is fully open). This is the value (with the opposite sign) to be set for the parameter `calibration5`.
```

File: docs/icub_r1_icub3_calibration_types/icub_r1_calibration_types.md [hunk 4]
```diff
@@ -168,6 +168,15 @@ This calibration is used for joint with absolute encoder, it is the newest versi
...
+After that, by observing how the values printed out by the port change, the user would check if the `invert direction`, i.e. `calibration3` value, should be `1` or `0` and if a specific value of rotation should be added at `calibration4` in order to avoid the encder to cross the zero while finger joint is moving.
```
Comment (pattacini) [inline docs/icub_r1_icub3_calibration_types/icub_r1_calibration_types.md:178]:
```suggestion
After that, while observing how the printed values change, you can check if the `invert direction` (i.e., the `calibration3` value), shall be `1` or `0`, and if a specific value of rotation shall be added up to `calibration4` to prevent the encoder from crossing the zero while finger joint is moving.
```

**Review by pattacini — CHANGES_REQUESTED** [review]
Some text steamline.
@MSECode, please double-check.

File: docs/icub_r1_icub3_calibration_types/icub_r1_calibration_types.md [hunk 5]
```diff
@@ -168,6 +168,15 @@ This calibration is used for joint with absolute encoder, it is the newest versi
...
 
+Aiming at helping the common user at finding the correct values for the paramters reported in the table below, which should be set in the calibration configuration file, the steps to follow for the identification of those values are here reported.
+First of all, the user should set the boolean values `skipCalibration` and `useRawEncoderData` in the file `general.xml` to `true`, therefore the calibration procedure won't start at `yarprobotinterface` startup and secondly it would be possible to allow the port `/ergocub/left_arm/state:o` to print the raw (i.e. uncalibrated values) for the finger joint encoders. 
+Done that, the user should check the values printed out by the port running in a terminal the command:
+
+`yarp read ... /ergocub/left_arm/state:o` 
+
```
Comment (pattacini) [inline docs/icub_r1_icub3_calibration_types/icub_r1_calibration_types.md:176]: Better off going with:

\```console
yarp read ... /ergocub/left_arm/state:o 
\```

**Review by pattacini — APPROVED** [review]

File: docs/icub_r1_icub3_calibration_types/icub_r1_calibration_types.md [hunk 6]
```diff
@@ -168,6 +168,16 @@ This calibration is used for joint with absolute encoder, it is the newest versi
 ### Calibration 14
 This calibration is used for joints with the absolute encoder at the joint (which currently works with POS service / FAP encoder sensor) and with incremental encoder at the motor. Moreover, it is applied to joints whose motor is provided with hard-stop mechanical limits. Specifically, this type of calibration is used when the user needs to reach one of the two mechanical limits on the joint motor, aiming at calibrating the incremental encoder at the motor and accurately setting the hardware limits between which the rotor can move.
 
+Find below some steps aimed to help you determine the correct values of the parameters to be set in the calibration sections of the configuration files.
+First, set to `true` the boolean variables `skipCalibration` and `useRawEncoderData` in the file `general.xml`. This is meant to disable calibration at `yarprobotinterface` startup. Also, this will make the port `/ergocub/left_arm/state:o` print the raw values (i.e., uncalibrated) for the finger joint encoders. 
+Once done, you should check the values printed out by the port running in a terminal the command:
+
+```console
+yarp read ... /ergocub/left_arm/state:o
+```
```
Comment (pattacini) [inline docs/icub_r1_icub3_calibration_types/icub_r1_calibration_types.md:177]: We need a blank line hereinafter.

File: docs/icub_r1_icub3_calibration_types/icub_r1_calibration_types.md [hunk 7]
```diff
@@ -168,6 +168,16 @@ This calibration is used for joint with absolute encoder, it is the newest versi
...
+and by doing this observing which should be the raw value of the offset for the encoder joint when the finger is at the hard-stop position (ideally finger full open). Thus, the user should set this value (with opposite sign if `invert direction` parameter is set to  `0`) as the value for the parameter `calibration5`.
+After that, by observing how the values printed out by the port change, the user would check if the `invert direction`, i.e. `calibration3` value, should be `1` or `0` and if a specific value of rotation should be added at `calibration4` in order to avoid the encder to cross the zero while finger joint is moving.
```
Comment (pattacini) [inline docs/icub_r1_icub3_calibration_types/icub_r1_calibration_types.md:179]: Hi @MSECode 

Would you mind harmonizing the text by, for example, replacing `user` with `you`?

**Review by pattacini — CHANGES_REQUESTED** [review]
I've still noticed something to streamline to improve reading.

File: docs/icub_r1_icub3_calibration_types/icub_r1_calibration_types.md (hunk 7 above)
Comment (pattacini) [reply docs/icub_r1_icub3_calibration_types/icub_r1_calibration_types.md:179]: Also,

```
avoid the encder to cross the zero while finger joint is moving.
```

should be

```
prevent the encoder from crossing the zero while finger joint is moving.
```

File: docs/icub_r1_icub3_calibration_types/icub_r1_calibration_types.md (hunk 6 above)
Comment (MSECode) [reply docs/icub_r1_icub3_calibration_types/icub_r1_calibration_types.md:177]: oks I'll fix that

File: docs/icub_r1_icub3_calibration_types/icub_r1_calibration_types.md [hunk 8]
```diff
@@ -166,7 +166,18 @@ This calibration is used for joint with absolute encoder, it is the newest versi
...
+
+Find below some steps aimed to help you determine the correct values of the parameters to be set in the calibration sections of the configuration files.
+First, set to `true` the boolean variables `skipCalibration` and `useRawEncoderData` in the file `general.xml`. This is meant to disable calibration at `yarprobotinterface` startup. Also, this will make the port `/ergocub/left_arm/state:o` print the raw values (i.e., uncalibrated) for the finger joint encoders. 
+Once done, you should check the values printed out by the port running in a terminal the command:
+
+```console
+yarp read ... /ergocub/left_arm/state:o
+```
+
+and by doing this observing which should be the raw value of the offset for the encoder joint when the finger is at the hard-stop position (ideally finger full open). Thus, you should set this value (with opposite sign if `invert direction` parameter is set to  `0`) as the value for the parameter `calibration5`.
```
Comment (pattacini) [inline docs/icub_r1_icub3_calibration_types/icub_r1_calibration_types.md:179]:
```
and by doing this observing which should be the raw value of the offset for the encoder joint when the finger is at the hard-stop position (ideally finger full open).
```

This part doesn't read well. Could you rephrase it?

Comment (MSECode) [reply docs/icub_r1_icub3_calibration_types/icub_r1_calibration_types.md:179]: sure thing

**Review by pattacini — APPROVED** [review]

File: docs/icub_r1_icub3_calibration_types/icub_r1_calibration_types.md (hunk 8 above)
Comment (pattacini) [reply docs/icub_r1_icub3_calibration_types/icub_r1_calibration_types.md]: Thanks!

**Comment by pattacini** [comment]
@MSECode 

I think the PR is mergeable now.
The PR is still in draft. Do you want me to wait for the FW PR's to be merged first?

**Review by pattacini — APPROVED** [review]
